mono/metadata/reflection.c

```diff
@@ -1550,9 +1550,9 @@ assembly_name_to_aname (MonoAssemblyName *assembly, char *p) {
 					p++;
 				}
 			}
-		} else if ((*p == 'P' || *p == 'p') && g_ascii_strncasecmp (p, "PublicKeyToken=", 15) == 0) {
+		} else if ((*p == 'P' || *p == 'p') && g_ascii_strncasecmp (p, G_STRING_CONSTANT_AND_LENGTH ("PublicKeyToken=")) == 0) {
 			p += 15;
-			if (strncmp (p, "null", 4) == 0) {
+			if (strncmp (p, G_STRING_CONSTANT_AND_LENGTH ("null")) == 0) {
```

Comment:
Build fails on Linux.  apparently `strncmp` is a macro and the preprocessor is unhappy with this line.

Reply:
Yeah, I just never got around to fixing this. I'm cleaning up various backlogged issues right now, will get to this eventually.

Reply:
Yeah, we discussed this, a while ago. To wrap it in g_strncmp or such.